### PR TITLE
[native pos] Refactor TestPrestoSparkHttpClient

### PR DESCRIPTION
**Pull request by mbasmanova**
Remove unused variables. Extract helper methods to reduce copy-paste.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java [hunk 1]
```diff
@@ -1027,10 +957,8 @@ else if (method.equalsIgnoreCase("DELETE")) {
                                     future.completeExceptionally(e);
                                 }
                             }
-                        }
-                        else if (method.equalsIgnoreCase("DELETE")) {
                             // DELETE /v1/task/{taskId}/results/{bufferId}
```
Comment (shrinidhijoshi) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java:960]: Nit: this and the comment on L:950 needs to exchanged.

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java [hunk 2]
```diff
@@ -1027,10 +957,8 @@ else if (method.equalsIgnoreCase("DELETE")) {
                                     future.completeExceptionally(e);
```
Comment (shrinidhijoshi) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java:957]: While you are at it, do you mind DRY'ing out this block as well into a function
```Java
try {
   future.complete(responseHandler.handle(
        request,
            responseManager.createResultResponse()));
}
catch (Exception e) {
      e.printStackTrace();
      future.completeExceptionally(e);
}
```

**Review by shrinidhijoshi — APPROVED** [review]
Thanks for improving this test @mbasmanova. A couple of nits.

**Review by pranjalssh — APPROVED** [review]
CI is failing, but accepting in advance

**Comment by ajaygeorge** [comment]
There are some checkstyle issues. Please fix them.
```
[INFO] There are 3 errors reported by Checkstyle 8.16 with src/checkstyle/presto-checks.xml ruleset.
Error:  src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java:[163,73] (blocks) LeftCurly: '{' at column 73 should be on a new line.
Error:  src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java:[175,102] (blocks) LeftCurly: '{' at column 102 should be on a new line.
Error:  src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java:[179,115] (blocks) LeftCurly: '{' at column 115 should be on a new line.
```

File: presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java (hunk 2 above)
Comment (mbasmanova) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/execution/http/TestPrestoSparkHttpClient.java]: Refactored by moving try-catch block outside of the if-else branches.

**Review by vermapratyush — APPROVED** [review]

**Pull request merged**